### DISCUSSION TITLE
Keycloak 21.1.1 hangs at start

### DISCUSSION THREAD

**Discussion by tobiasge** [keycloak/keycloak#20676] [Q&A]
Hello,
I'm having trouble with my Keycloak start up. It just stops doing anything and when I attach a debugger I can see that most of the threads are waiting. But I don't know for what exactly. I suspect it has something to do with the cache.
At the moment I start only one instance of Keycloak, but later I want to have two on two servers.

Here are my configuration and logs:
https://gist.github.com/tobiasge/ada82b9d31298630814362c99601c15c

I'm starting Keycloak with this command: `bash -c 'source ./env; ./bin/kc.sh start --debug --cache-config-file cache-ispn.xml'`
After it hangs Ctrl+C can't stop the Keycloak process; only a `kill -9` helps.

I hope somebody has an idea what's going on here.

**Comment by bernhof** [comment]
When using `bash -c ...` docker can't capture SIGINT at stop the container on Ctrl+C.

Instead, use command `start --debug --cache-config-file cache-ispn.xml` and set environment variables using `docker run -e VAR1=something` ([docs](https://docs.docker.com/reference/cli/docker/container/run/#env)) or docker compose's `environment` ([docs](https://docs.docker.com/compose/compose-file/05-services/#environment))
